lib/fw.c

```diff
@@ -1292,6 +1294,11 @@ int switchtec_fw_is_boot_ro(struct switchtec_dev *dev)
 
 	int ret;
 
+	if (!switchtec_is_gen3(dev)) {
+		errno = ENOTSUP;
+		return -1;
```

Comment:
Minor fix: Should probably just return 0 saying the partition is not read only when it's not supported.

Reply:
F/W is still updating this part of MRPC interface, in gen4, it's more likely that we will have no MRPC command to set RW to a partition and then allow for programming it. So let's wait and see what will be final interface from the F/W and tuning this part in the software accordingly.